storage-proofs/src/parameter_cache.rs

```diff
@@ -106,6 +106,46 @@ where
             None => Ok(generate()?),
         }
     }
+
+    fn get_verifying_key(circuit: C, pub_params: &PP) -> Result<groth16::VerifyingKey<E>> {
+        let generate = || -> Result<groth16::VerifyingKey<E>> {
+            let groth_params = Self::get_groth_params(circuit, pub_params)?;
+            info!(SP_LOG, "Getting verifying key."; "target" => "verifying_key");
+            Ok(groth_params.vk)
+        };
+
+        match Self::cache_identifier(pub_params) {
```

Comment:
as far as I can tell `cache_identifier` always returns `Some`, can we just stop returning an `Option` from it?

Reply:
It was there so required implementations could opt out of caching if needed, but in practice it hasn't been needed, so … goodbye.